gui/default/syncthing/core/syncthingController.js

```diff
@@ -563,9 +578,53 @@ angular.module('syncthing.core')
                 }
                 $scope.connections = data;
                 console.log("refreshConnections", data);
+
+                refreshRemoteGUI(data);
             }).error($scope.emitHTTPError);
         }
 
+        function refreshRemoteGUI(connections) {
+            if (!$scope.showRemoteGUI) {
+                $scope.idToRemoteGUI = {}
+                return
+            }
+            var newCache = {};
+            for (var id in connections) {
+                if (!connections.hasOwnProperty(id)) {
+                    continue;
+                }
+                if (!(id in $scope.devices)) {
+                    // Avoid errors when called before first updateLocalConfig()
+                    continue;
+                }
+                if (!(id in $scope.idToRemoteGUI)) {
+                    $scope.idToRemoteGUI[id] = "";
+                }
+                var port = $scope.devices[id].remoteGUIPort;
+                var isNotRelayConnection = !connections[id].type.includes("relay");
+                if (connections[id].address != "" && isNotRelayConnection && port > 0) {
+                    var newAddress = "http://" + replaceAddressPort(connections[id].address, port);
+                    if (!(newAddress in $scope.remoteGUICache)) {
+                        // No cached result, trigger a new port probing asynchronously
+                        $scope.probeRemoteGUIAddress(id, newAddress);
+                    } else {
+                        newCache[newAddress] = $scope.remoteGUICache[newAddress];
+                        // Copy cached probing result in the corner case of duplicate GUI
+                        // addresses for different devices.  Which is useless, but
+                        // possible when behind the same NAT router.
+                        if (newCache[newAddress]) {
+                            $scope.idToRemoteGUI[id] = newAddress;
+                        } else {
+                            $scope.idToRemoteGUI[id] = "";
+                        }
+                    }
+                }
+            }
+            // Replace the cache to discard stale addresses
+            $scope.remoteGUICache = newCache;
+            console.log("refreshRemoteGUI");
```

Comment:
That doesn't add much value? If we fear this might be very slow, then a pair of log statements like `starting refreshRemoteGUI` and here `starting refreshRemoteGUI` would give an indication in the console of how long it took.

Reply:
I just thought it was customary as many other `refresh...()` functions print out their name and some data.  Removed.